packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -236,6 +236,53 @@ export default function parse(source: string, state: State): TemplateParseResult
                 }
             },
         },
+
+        Comment: {
+            enter(node: parse5.AST.Default.CommentNode) {
+                const location = node.__location as parse5.MarkupData.Location;
+                const rawComment = node.data;
+                const tokenizedContent = rawComment.split(EXPRESSION_RE);
+
+                const commentParts = [];
+                const commentDynamicParts = [];
+                let currentPartsIndex = 0;
+
+                for (const token of tokenizedContent) {
```

Comment:
I don't think we should add binding support for comments. I would argue that if you need to render some dynamic content in your comment, you have to use the `lwc:inner-html` directive.

Reply:
well, that's debatable, and it's great that you bring it ... taking notes for the rfc. at first glance: I like the ergonomics of it, by doing it `lwc:inner-html` you will need to add a new element just to add the comment, how will that work when you have in a template, some part that needs to be wrapped in mso-comments?